app/assets/javascripts/dashboard/dashboard.js

```diff
@@ -508,6 +566,30 @@ function removePendingAcceptance(minionId) {
   sessionStorage.removeItem(minionId);
 }
 
+function hasPendingRejection(minionId) {
+  return sessionStorage.getItem('reject-'+minionId) === 'true';
+}
+
+function setPendingRejection(minionId) {
+  sessionStorage.setItem('reject-'+minionId, true);
+  $('.pending-nodes-container a[data-minion-id="' + minionId + '"]').parent().text('Rejection in progress');
+}
+
+function removePendingRejection(minionId) {
+  sessionStorage.removeItem('reject-'+minionId);
+}
+
+function setPendingRemoval(minionId) {
+  $('.nodes-container a[data-minion-id="' + minionId + '"]').parent().text('Pending removal');
+  $('#role-column-' + minionId + ' .btn').prop('disabled', true);
```

Comment:
After testing I was able to identify that there was some delay disabled the other rows buttons, so we can safely replace this selector with `.role-column .btn` like

```js
$('.role-column .btn').prop('disabled', true);
```